src/main/java/uk/gov/hmcts/juror/api/moj/service/report/YieldPerformanceReportServiceImpl.java

```diff
@@ -0,0 +1,186 @@
+
+        YieldPerformanceReportResponse response = setupResponseHeaders(yieldPerformanceReportRequest);
+
+        List<String> courtLocCodesList = yieldPerformanceReportRequest.isAllCourts()
+            ? courtQueriesRepository.getAllCourtLocCodes(false)
+            : yieldPerformanceReportRequest.getCourtLocCodes();
+
+        String courtLocCodes = String.join(",", courtLocCodesList);
+
+        try {
+            List<String> yieldPerformanceReportStats =
+                jurorPoolRepository.getYieldPerformanceReportStats(courtLocCodes,
+                    yieldPerformanceReportRequest.getFromDate(),
+                    yieldPerformanceReportRequest.getToDate());
+
+            List<Tuple> poolComments = poolCommentRepository.findPoolCommentsForLocationsAndDates(
+                courtLocCodesList,
+                yieldPerformanceReportRequest.getFromDate(),
+                yieldPerformanceReportRequest.getToDate());
+
+            Map<String, StringBuilder> poolCommentsMap = getPoolsCommentsMap(poolComments);
+
+            setupResponseDto(response, yieldPerformanceReportStats, poolCommentsMap);
+
+        } catch (Exception e) {
+            log.error("Error getting yield performance report by court", e);
+            throw new MojException.InternalServerError("Error getting yield performance report by court", e);
+        }
+
+        return response;
+    }
+
+    private static Map<String, StringBuilder> getPoolsCommentsMap(List<Tuple> poolComments) {
+        Map<String, StringBuilder> poolCommentsMap = new HashMap<>();
+
+        for (Tuple poolComment : poolComments) {
+            poolCommentsMap.computeIfPresent(poolComment.get(0, String.class),
+                (key, val) -> val.append("\r\n").append(poolComment.get(1, String.class))
+                    .append(" - ").append(poolComment.get(2, String.class)));
+            poolCommentsMap.putIfAbsent(poolComment.get(0, String.class), new StringBuilder(poolComment.get(1,
+                String.class)).append(" - ").append(poolComment.get(2, String.class)));
+        }
+        return poolCommentsMap;
+    }
+
+    private void setupResponseDto(YieldPerformanceReportResponse response, List<String> yieldPerformanceReportStats,
+                                  Map<String, StringBuilder> poolCommentsMap) {
+        if (yieldPerformanceReportStats != null && !yieldPerformanceReportStats.isEmpty()) {
+            for (String result : yieldPerformanceReportStats) {
+                List<String> values = List.of(result.split(","));
+                setupResponse(response, values, poolCommentsMap);
+            }
+        }
+    }
+
+    private void setupResponse(YieldPerformanceReportResponse response, List<String> result,
+                               Map<String, StringBuilder> poolCommentsMap) {
+
+        int requested = Integer.parseInt(result.get(2));
+        int confirmed = Integer.parseInt(result.get(3));
```

Comment:
Same as above result.get(2) does not provide any context would be good to have result as an object

Reply:
updated

Reply:
updated